src/code-considerations/using-unstable-lang/const-evaluation.md

```diff
@@ -0,0 +1,100 @@
+# Const evaluation
+
+Rust's compile-time code evaluation story is still in an experimental state,
+with a large amount of unstable surface area, including areas where there is
+active design work.
+
+The standard library (as of this writing) uses some amount of hacks to work
+around language limitations for some implementations: this document provides a
+framework for evaluating the cost/benefit analysis of using such workarounds to
+enable adding new `const` functions/traits or `const`-ifying existing APIs.
+
+The overall high-level guidance is to evaluate the tradeoff between enabling
+experimentation in users of the standard library (or discovery of limitations
+to these language features) and the costs to the maintenance and readability of
+the standard library. (This is broad guidance applicable to almost any feature,
+but const generics are particularly notable in terms of their potentially wide
+scope).
+
+These are good areas to consider in such an evaluation. You should think about
+whether the given PR is adding value to outweigh its costs along these
+dimensions, and add to the PR description what your justification for the usage
+is. Consider including future plans for the feature as well: if the feature is
+experimental, does it have a clear path toward stabilization in the current
+form?
+
+## Maintainability
+
+Does making this API const require that code be hand-written which was previously automatic?
+
+Does the new code require duplicating, for example through `#[cfg(bootstrap)]`,
+increasing the risk of diverging between the two implementations (and
+increasing work when bumping the bootstrap compiler)?
+
+Will updates/bugfixes to the code require further workarounds to CTFE
+limitations?
+
+Are the changes making a decision for many parts of the standard library?  For
+example, if a trait is made const, that likely means all implementations in the
+standard library have to be const too. It may not be the case that this is easy
+to accomplish (particularly if their current implementations are complex and
+need const hacks to work around limitations in CTFE).
+
+## Stability risks
+
+When making an internal (or external) API `const`, it becomes harder to
+evaluate whether a given standard library function is safe to stabilize as
+const. One of its dependencies may take a long time to stabilize for
+const-evaluation, even though its API doesn't obviously reference unstable
+surface area. 
```

Comment:
I am a bit confused by this point here. If an API isn't `const` in the first place, then we don't need to evaluate whether it is okay to stabilize as `const`. If it is, then we can always split the feature up if unsure.